lib/Conversion/TTIRToTTIRDecomposition/TTIRToTTIRDecomposition.cpp

```diff
@@ -283,14 +287,149 @@ struct ConvolutionToConv2dPattern
     return success();
   }
 
-  LogicalResult
-  matchAndRewrite(ttir::ConvolutionOp op, OpAdaptor adaptor,
-                  ConversionPatternRewriter &rewriter) const override {
+  ttir::ReshapeOp createReshapeOp(Location loc, Value tensor,
+                                  llvm::ArrayRef<int64_t> target_input_shape,
+                                  ::mlir::ArrayAttr constraints,
+                                  ConversionPatternRewriter &rewriter) const {
+    auto inputType = mlir::cast<RankedTensorType>(tensor.getType());
+
+    auto output = rewriter.create<tensor::EmptyOp>(
+        loc, llvm::ArrayRef<int64_t>(target_input_shape),
+        inputType.getElementType());
+    std::vector<int32_t> shapei32(target_input_shape.begin(),
+                                  target_input_shape.end());
+    auto shape_attr = rewriter.getI32ArrayAttr(shapei32);
+
+    return rewriter.create<ttir::ReshapeOp>(
+        loc,
+        mlir::RankedTensorType::get(target_input_shape,
+                                    inputType.getElementType()),
+        tensor, output, shape_attr, constraints);
+  }
+
+  mlir::DenseI64ArrayAttr
+  addIntegersToDenseArrayAttr(ConversionPatternRewriter &rewriter,
+                              mlir::DenseI64ArrayAttr denseArrayAttr,
+                              std::vector<uint64_t> additionalIntegers) const {
+    llvm::SmallVector<int64_t, 4> newDenseArray(denseArrayAttr.asArrayRef());
+    for (uint64_t integer : additionalIntegers) {
+      newDenseArray.push_back(integer);
+    }
+    return rewriter.getDenseI64ArrayAttr(newDenseArray);
+  }
+
+  mlir::DenseBoolArrayAttr
+  addBooleansToDenseArrayAttr(ConversionPatternRewriter &rewriter,
+                              mlir::DenseBoolArrayAttr denseArrayAttr,
+                              std::vector<bool> additionalIntegers) const {
+    llvm::SmallVector<bool, 4> newDenseArray(denseArrayAttr.asArrayRef());
+    for (bool boolean : additionalIntegers) {
+      newDenseArray.push_back(boolean);
+    }
+    return rewriter.getDenseBoolArrayAttr(newDenseArray);
+  }
 
-    if (failed(isConv2d(op))) {
+  LogicalResult matchAndRewrite1d(ttir::ConvolutionOp op, OpAdaptor adaptor,
+                                  ConversionPatternRewriter &rewriter) const {
+    // Not currently supporting spatial dims other than 2 for the 1D case.
+    if (op.getConvolutionLayout().getInputSpatialDimensions()[0] != 2) {
       return failure();
     }
 
+    auto outputType =
+        mlir::cast<RankedTensorType>(adaptor.getOutput().getType());
+    llvm::ArrayRef<int64_t> outputShape = outputType.getShape();
+    llvm::SmallVector<int64_t, 8> newOutputShape(outputShape.begin(),
+                                                 outputShape.end());
+    newOutputShape.push_back(1);
```

Comment:
There are a lot of these pushes without any comments, I don't understand what they're for.